pdl_syntax.md

```diff
@@ -0,0 +1,828 @@
+Each member can be primitive, record, enum map or array type.
+
+For example :
+```
+namespace com.linkedin.pegasus.generator.test.idl.unions
+
+record WithPrimitivesUnion {
+  `union`: union[int, long, float, double, boolean, string, bytes]
+}
+```
+
+The member type names also serve as the “member keys” (also called as “union tags”), and identify which union member type data holds.
+To define a field of a record containing a union of two other records, we would define:
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Question {
+  answerFormat: union[MultipleChoice, TextEntry]
+}
+```
+
+### Union with aliases
+Union members can optionally be given an alias. Aliases can be used to create unions with members of the same type or to give better naming for union members.
+
+Aliased unions would be defines as :
+```
+union [alias: type, ...]
+```
+
+For example :
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Question {
+ answerFormat: union[   
+   multipleChoice: MultipleChoice,
+   shortAnswer: string,
+   longAnswer: string
+ ]
+}
+```
+In the above example, the union answerFormat has three members, with two string type members differentiated using the aliases (shortAnswer and longAnswer). When aliases are used, the alias becomes the "member key" for the union members and will be used in the wire format. 
+
+## Enum Type
+
+Enums types may contain any number of symbols. 
+
+For example :
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+enum Fruits {
+  APPLE
+  BANANA
+  ORANGE
+  PINEAPPLE
+}
+```
+
+Enums can be referenced in other schemas by name.
+
+For example :
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record FruitBasket {
+  fruit: Fruits
+}
+```
+
+Enums can also be referenced in other schemas by inlining their type definition. 
+
+For example :
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record FruitBasket {
+  fruit: enum Fruits { APPLE, BANANA, ORANGE }
+} 
+```
+
+### Enum documentation, deprecation and properties
+Doc comments, @deprecation and properties can be added directly to enum symbols. 
+
+For example :
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * A fruit
+ */
+enum Fruits {
+
+  @color = "red"
+  APPLE
+
+  /**
+   * A yummy fruit.
+   */
+  @color = "yellow"
+  BANANA
+
+  @deprecated
+  @color = "orange"
+  ORANGE
+}
+```
+
+### Enum defaults
+To specify defaults, specify the enum value as string.
+
+For example :
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record FruitBasket {
+  fruit: enum Fruits { APPLE, BANANA, ORANGE } = "APPLE"
+} 
+```
+
+
+The default value can also be defined as following exmaple:
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record FruitBasket {
+  fruit: Fruits = "APPLE"
+} 
+```
+
+## Typerefs
+Pegasus supports a new schema type known as a typeref. A typeref is like
```

Comment:
Like we discussed, we need to keep the details here to be specific to the syntax. Just use a simple sentence to explain what a typeref is and remove the other details.

Reply:
Updated.